### PR TITLE
[RFR] ES6 and adapters

### PR DESCRIPTION
**Pull request by djhi**
- [x] rewritten using es6 with babel
- [x] added eslint air-bnb configuration
- [x] introduced adapters (redis and memory)
- [x] updated README

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    *
+    * @param {Function} fn
+    * @api public
+    *//**
+    * Inspect implementation.
+    *
+    * @api public
+    */
+    inspect() {
+        return `<Limiter id='${this.id}', duration='${this.duration}', max='${this.max}'>`;
```
Comment (Kmaschta) [inline src/index.js]: There is a way to display the adapter here?

Comment (djhi) [inline src/index.js]: Unless we add a `name` property to adapters, I don't think so

Comment (Kmaschta) [inline src/index.js]: Even with `adapter.name` ? http://www.2ality.com/2015/09/function-names-es6.html

File: History.md [hunk 2]
```diff
@@ -1,5 +1,10 @@
 # ratelimiter Change Log.
 
+## 2.0.0 / 2016-03-01
+* rewritten using es6 with babel
+* added eslint air-bnb configuration
```
Comment (Kmaschta) [inline History.md]: There is no `-` in airbnb

File: Readme.md [hunk 3]
```diff
@@ -31,7 +31,7 @@ $ npm install ratelimiter
 
 ```js
 var id = req.user._id;
-var limit = new Limiter({ id: id, db: db });
+var limit = new Limiter({ id: id }, redisAdapter(db)));
```
Comment (Kmaschta) [inline Readme.md]: `redisAdapter(redis.createClient())` ?

File: Readme.md [hunk 4]
```diff
@@ -58,11 +58,33 @@ limit.get(function(err, limit){
...
  - `duration` - of limit in milliseconds [3600000]
 
+# Adapters
+
+## RedisAdapter
+
+Initialize a new adapter with:
+
+```js
+var adapter = redisAdapter(db);
```
Comment (Kmaschta) [inline Readme.md]: idem

File: lib/adapters/memory.js [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+                    if (data.reset && dateNow / 1000 > data.reset) {
+                        return resolve(reset(data, dateNow));
+                    }
+
+                    resolve(data);
+                });
+            }
+        };
+    };
+};
```
Comment (Kmaschta) [inline lib/adapters/memory.js]: Space at end of file

File: Readme.md (hunk 3 above)
Comment (djhi) [inline Readme.md]: previous samples in README assumed `db` was a correctly initiated redis client.

**Comment by Kmaschta** [comment]
There is a specific reason to pass `duration` and `max` at each new hit instead of pass options to the adapter factory ?

**Comment by djhi** [comment]
We don't pass them at each new hit

**Comment by Kmaschta** [comment]
Yup, I deleted the comment.

Great job!

File: Readme.md (hunk 3 above)
Comment (Kmaschta) [inline Readme.md]: Explicit is better than implicit, IMO

**Pull request merged**